Source_Code/Client/ClueLessClient.java

```diff
@@ -62,29 +67,33 @@ public static void main(String[] args)
         clientApplication.setPlayerName(name);
         // BEGIN TEMPORARY - Just to let us test out the interface
         // Read user input from CLI and perform appropriate action
-        int input;
-        do
+        int input = 0;
+        
+        while(input != -1 && !clientApplication.activeGame)
         {
             System.out.println("****Enter a command****");
             System.out.println("Send action request to server: 1");
             System.out.println("See total num received status updates: 2");
             System.out.println("Send Connect Request: 3");  // To test sending multiple
             System.out.println("Request Start Game: 4");
             System.out.println("Exit: -1");
-            input = scan.nextInt();
+
+            input = Integer.parseInt( reader.read() );
```

Comment:
I'm pretty sure this is the source of the input bug. I need to come up with a way to keep the reader resource from being used by the main thread, otherwise the user input that's supposed to be meant for the suggest logic is getting claimed here (and causing a need to enter it twice).